pkg/controller/component/component_resources.go

```diff
@@ -45,17 +52,41 @@ func newOutputImageStream(cr *componentsv1alpha1.Component) *imagev1.ImageStream
 	}}
 }
 
-func newBuildConfig(cr *componentsv1alpha1.Component, builder *imagev1.ImageStream) *buildv1.BuildConfig {
+func (r *ReconcileComponent) newBuildConfig(cr *componentsv1alpha1.Component, builder *imagev1.ImageStream, gitSource *gitsourcev1alpha1.GitSource) *buildv1.BuildConfig {
 	labels := resource.GetLabelsForCR(cr)
 	buildSource := buildv1.BuildSource{
 		Git: &buildv1.GitBuildSource{
-			URI: cr.Spec.Codebase,
-			Ref: "master",
+			URI: gitSource.Spec.URL,
+			Ref: gitSource.Spec.Ref,
 		},
 		Type: buildv1.BuildSourceGit,
 	}
 	incremental := true
 
+	// Check if secrets provided exist or not
+	if gitSource.Spec.SecretRef != nil && gitSource.Spec.SecretRef.Name != "" {
+		u := unstructured.Unstructured{}
+		u.SetGroupVersionKind(schema.GroupVersionKind{
+			Kind:    "Secret",
+			Version: "v1",
+			Group:   "",
+		})
+
+		err := r.client.Get(context.TODO(), client.ObjectKey{
+			Namespace: cr.Namespace,
+			Name:      gitSource.Spec.SecretRef.Name,
+		}, &u)
+		if err != nil {
+			// TODO(Akash) - Need to discuss if we need to create buildConfig if secret isn't present.
+			log.Error(err, fmt.Sprintf("Failed to get provided secret please check if secrets %s present or not", gitSource.Spec.SecretRef.Name))
```

Comment:
The secret would be present only if the user is using private repos. It would be absent otherwise.

Reply:
Ack